src/controllers/quizController.js

```diff
@@ -0,0 +1,73 @@
+const quizModel = require("../models/quizSchema");
+const { title, category, level, label, questions } = require("../db/data");
+
+// questions controllers
+
+/** insert  question */
+const insertQuiz = async (req, res) => {
+  const { role } = req.user;
+
+  try {
+    if (role !== "admin") {
+      return res.status(403).json({
+        message: "Only Admin can add questions",
+      });
+    }
+    const quiz = await quizModel.insertMany({
+      title,
+      category,
+      level,
+      label,
+      questions,
+    });
+    res.status(200).json({
+      message: "Data Saved Sucessfuly",
+      quiz,
+    });
+  } catch (error) {
+    res.status(400).json(error.message);
+  }
+};
+
+/** get all quiz */
+const getAllQuiz = async (req, res) => {
+  try {
+    const quiz = await quizModel.find();
+    res.status(200).json(quiz);
+  } catch (error) {
+    res.status(400).json(error.message);
+  }
+};
+
+/** get quiz per  choice(category,level...) */
+const getQuizPerChoice = async (req, res) => {
```

Comment:
As i said i saw your review too late Sir so if you can accept this because frontend team has already worked with this and make PRs with that .Otherwise I agree with you